packages/tasks/package.json

```diff
@@ -23,8 +23,8 @@
 		"lint:check": "eslint --ext .cjs,.ts .",
 		"format": "prettier --write .",
 		"format:check": "prettier --check .",
-		"prepublishOnly": "pnpm run build",
-		"build": "tsup src/index.ts --format cjs,esm --clean --dts && pnpm run inference-codegen",
+		"prepublishOnly": "pnpm run inference-codegen && git diff --name-only --exit-code src && pnpm run build",
+		"build": "tsup src/index.ts --format cjs,esm --clean && tsc --emitDeclarationOnly --declaration",
```

Comment:
@Wauplin @SBrandeis I changed the codegen to automatically run in `@huggingface/tasks` only during prepublish (publish will error if not up to date)

We would do the same for tgi codegen

It avoids doing it during every `pnpm install` but should be done manually before publishing